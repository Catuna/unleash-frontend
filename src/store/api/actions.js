import api from '../../data/api';

export const RECIEVE_API_DETAILS = 'RECIEVE_API_DETAILS';
export const ERROR_RECIEVE_API_DETAILS = 'ERROR_RECIEVE_API_DETAILS';

export const RECEIVE_APPLICATION = 'RECEIVE_APPLICATION';

const recieveApiDetails = json => ({
    type: RECIEVE_API_DETAILS,
    value: json,
});

const errorRecieveApiDetails = (statusCode, type = ERROR_RECIEVE_API_DETAILS) => ({
    type,
    statusCode,
});

export function fetchAll() {
    return dispatch =>
        api
            .fetchAll()
            .then(json => dispatch(recieveApiDetails(json)))
            .catch(error => dispatch(errorRecieveApiDetails(error)));
}
