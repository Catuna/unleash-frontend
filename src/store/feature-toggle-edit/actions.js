const debug = require('debug')('unleash:feature-toggle-edit-actions');

export const TOGGLE_EDIT_INIT = 'TOGGLE_EDIT_INIT';
export const TOGGLE_EDIT_CLEAR = 'TOGGLE_EDIT_ClEAR';
export const TOGGLE_EDIT_TOUCH = 'TOGGLE_EDIT_TOUCH';
export const TOGGLE_EDIT_UPDATE = 'TOGGLE_EDIT_UPDATE';
export const TOGGLE_EDIT_ADD_GROUP = 'TOGGLE_EDIT_ADD_GROUP';
export const TOGGLE_EDIT_REMOVE_STRATEGY = 'TOGGLE_EDIT_REMOVE_STRATEGY';
export const TOGGLE_EDIT_ADD_STRATEGY_TO_GROUP = 'TOGGLE_EDIT_ADD_STRATEGY_TO_GROUP';

export function init(featureToggle) {
    debug('init ', featureToggle.name);
    return { type: TOGGLE_EDIT_INIT, featureToggle };
}

export function clearEdit(featureToggleName) {
    debug('clear ', featureToggleName);
    return { type: TOGGLE_EDIT_CLEAR, featureToggleName };
}

export function touch(featureToggleName) {
    debug('touch ', featureToggleName);
    return { type: TOGGLE_EDIT_TOUCH, featureToggleName };
}

export function update(featureToggle) {
    debug('Update ', featureToggle.name);
    return { type: TOGGLE_EDIT_UPDATE, featureToggle };
}

export function addGroup(featureToggleName) {
    debug('Add group ', featureToggleName);
    return {
        type: TOGGLE_EDIT_ADD_GROUP,
        featureToggleName,
    };
}

export function removeStrategy(featureToggleName, index) {
    debug(`Remove strategy ${index} from ${featureToggleName}`);
    return {
        type: TOGGLE_EDIT_REMOVE_STRATEGY,
        featureToggleName,
        index,
    };
}

export function addStrategyToGroup(featureToggleName, groupIndex, strategy) {
    debug(`Add strategy to group ${groupIndex} for ${featureToggleName}`);
    return {
        type: TOGGLE_EDIT_ADD_STRATEGY_TO_GROUP,
        featureToggleName,
        groupIndex,
        strategy,
    };
}
