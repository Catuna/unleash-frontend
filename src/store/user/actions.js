export const USER_UPDATE_USERNAME = 'USER_UPDATE_USERNAME';
export const USER_SAVE = 'USER_SAVE';

export const updateUserName = (value) => ({
    type: USER_UPDATE_USERNAME,
    value,
});

export const save = () => ({
    type: USER_SAVE,
});
