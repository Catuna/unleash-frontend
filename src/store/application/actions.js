import api from '../../data/applications-api';

export const RECEIVE_APPLICATINS = 'RECEIVE_APPLICATINS';
export const ERROR_RECEIVE_APPLICATINS = 'ERROR_RECEIVE_APPLICATINS';

const recieveApplications = (json) => ({
    type: RECEIVE_APPLICATINS,
    value: json,
});

const errorReceiveApplications = (statusCode) => ({
    type: ERROR_RECEIVE_APPLICATINS,
    statusCode,
});

export function fetchApplications () {
    return dispatch => api.fetchAll()
        .then(json => dispatch(recieveApplications(json)))
        .catch(error => dispatch(errorReceiveApplications(error)));
}
