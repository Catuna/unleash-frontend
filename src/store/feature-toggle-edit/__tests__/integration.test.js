import { AssertionError } from 'assert';
import reducer from '../index';
import { init, clearEdit, addGroup, removeStrategy, addStrategyToGroup } from '../actions';

const name = 'test';

const featureToggle = {
    name,
    strategies: [
        {
            name: 'default',
        },
    ],
};

const initalizedState = reducer(undefined, init(featureToggle));

test('should be inital state', () => {
    const state = reducer(undefined, {});
    expect(state.toJS()).toEqual({});
});

test('should initialize toggle', () => {
    const action = init(featureToggle);
    const state = reducer(undefined, action);
    expect(state.get(name).get('name')).toBe(name);
    expect(state.get(name).toJS()).toMatchSnapshot();
});

test('should clear toggle from store', () => {
    const initalState = reducer(undefined, init(featureToggle));
    const state = reducer(initalState, clearEdit(name));
    expect(state.get(name)).toBeUndefined();
});

test('should add group to toggle', () => {
    const initalState = reducer(undefined, init(featureToggle));
    const state = reducer(initalState, addGroup(name));
    const jsToggle = state.get(name).toJS();
    expect(jsToggle.dirty).toBeTruthy();
    expect(jsToggle).toMatchSnapshot();
});

test('should remove strategy with given index', () => {
    const stateWithGroup = reducer(initalizedState, addGroup(name));
    const state = reducer(stateWithGroup, removeStrategy(name, 1));

    const jsToggle = state.get(name).toJS();
    expect(jsToggle.strategies.length).toBe(1);
    expect(jsToggle.dirty).toBeTruthy();
    expect(jsToggle).toMatchSnapshot();
});

test('should remove second strategy', () => {
    const toggle = {
        name,
        strategies: [{ name: 'first' }, { name: 'second' }, { name: 'third' }],
    };

    const initState = reducer(undefined, init(toggle));
    const state = reducer(initState, removeStrategy(name, 1));

    const jsToggle = state.get(name).toJS();
    expect(jsToggle.strategies.length).toBe(2);
    expect(jsToggle.dirty).toBeTruthy();
    expect(jsToggle).toMatchSnapshot();
});

test('should add strategy to group', () => {
    const stateWithGroup = reducer(initalizedState, addGroup(name));
    const state = reducer(stateWithGroup, addStrategyToGroup(name, 1, { name: 'default' }));

    const jsToggle = state.get(name).toJS();
    expect(jsToggle.strategies.length).toBe(2);
    expect(jsToggle.strategies[1].group.length).toBe(1);
    expect(jsToggle.dirty).toBeTruthy();
    expect(jsToggle).toMatchSnapshot();
});

test('should add multiple strategies to group', () => {
    const s0 = reducer(initalizedState, addGroup(name));
    const s1 = reducer(s0, addStrategyToGroup(name, 1, { name: 'default' }));
    const s2 = reducer(s1, addStrategyToGroup(name, 1, { name: 'custom' }));
    const s3 = reducer(s2, addStrategyToGroup(name, 1, { name: 'foo' }));

    const jsToggle = s3.get(name).toJS();
    expect(jsToggle.strategies.length).toBe(2);
    expect(jsToggle.strategies[1].group.length).toBe(3);
    expect(jsToggle.dirty).toBeTruthy();
    expect(jsToggle).toMatchSnapshot();
});

test('should not be able to add strategy to no group', () => {
    const f = () => reducer(initalizedState, addStrategyToGroup(name, 0, { name: 'default' }));
    expect(f).toThrow(AssertionError);
});
