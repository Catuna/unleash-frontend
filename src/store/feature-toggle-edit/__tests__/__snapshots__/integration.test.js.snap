// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add group to toggle 1`] = `
Object {
  "dirty": true,
  "name": "test",
  "strategies": Array [
    Object {
      "name": "default",
    },
    Object {
      "group": Array [],
      "name": "__internal-strategy-group",
    },
  ],
}
`;

exports[`should add multiple strategies to group 1`] = `
Object {
  "dirty": true,
  "name": "test",
  "strategies": Array [
    Object {
      "name": "default",
    },
    Object {
      "group": Array [
        Object {
          "name": "default",
        },
        Object {
          "name": "custom",
        },
        Object {
          "name": "foo",
        },
      ],
      "name": "__internal-strategy-group",
    },
  ],
}
`;

exports[`should add strategy to group 1`] = `
Object {
  "dirty": true,
  "name": "test",
  "strategies": Array [
    Object {
      "name": "default",
    },
    Object {
      "group": Array [
        Object {
          "name": "default",
        },
      ],
      "name": "__internal-strategy-group",
    },
  ],
}
`;

exports[`should initialize toggle 1`] = `
Object {
  "name": "test",
  "strategies": Array [
    Object {
      "name": "default",
    },
  ],
}
`;

exports[`should remove second strategy 1`] = `
Object {
  "dirty": true,
  "name": "test",
  "strategies": Array [
    Object {
      "name": "first",
    },
    Object {
      "name": "third",
    },
  ],
}
`;

exports[`should remove strategy with given index 1`] = `
Object {
  "dirty": true,
  "name": "test",
  "strategies": Array [
    Object {
      "name": "default",
    },
  ],
}
`;
